tests/v2/validation/rbac/clusterandprojectroles/cluster_role_test.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RBACAdditionalTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (rb *RBACAdditionalTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *RBACAdditionalTestSuite) SetupSuite() {
+	rb.session = session.NewSession()
+
+	client, err := rancher.NewClient("", rb.session)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in rb")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *RBACAdditionalTestSuite) TestClusterOwnerAddsUserAsProjectOwner() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterAdmin, clusterAdminClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	additionalUser, additionalUserClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Adding user as " + rbac.ClusterOwner.String() + " to the downstream cluster.")
+	err = users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.ClusterOwner.String(), nil)
+	require.NoError(rb.T(), err)
+	clusterAdminClient, err = clusterAdminClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	clusterOwnerProject, err := clusterAdminClient.Management.Project.Create(projects.NewProjectConfig(rb.cluster.ID))
+	require.NoError(rb.T(), err)
+	err = users.AddProjectMember(clusterAdminClient, clusterOwnerProject, additionalUser, rbac.ProjectOwner.String(), nil)
+	require.NoError(rb.T(), err)
+
+	projectListClusterAdditionalUser, err := projects.ListProjectNames(additionalUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	assert.Equal(rb.T(), 1, len(projectListClusterAdditionalUser))
+	assert.Equal(rb.T(), clusterOwnerProject.Name, projectListClusterAdditionalUser[0])
+
+	err = users.RemoveProjectMember(clusterAdminClient, additionalUser)
+	require.NoError(rb.T(), err)
+	projectListClusterAdditionalUser, err = projects.ListProjectNames(additionalUserClient, rb.cluster.ID)
+	require.Empty(rb.T(), projectListClusterAdditionalUser)
+}
+
+func (rb *RBACAdditionalTestSuite) TestClusterOwnerAddsUserAsClusterOwner() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterAdmin, clusterAdminClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	errUserRole := users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.ClusterOwner.String(), nil)
```

Comment:
what is the purpose of this user? it isn't consistent with the previous test, nor the test title imo

Reply:
this is to add clusterAdmin created in step 85 to the cluster as a cluster owner. Since we are not using this var anywhere other than for error check and the returned arg is of type error, we named this as errUserRole.

Reply:
IMO 85 - 90 should be put in a helper function. Its reused in these tests and appears to be 'create the initial user'. 

mentioned in a separate comment but specifically naming the `err` to something other than `err` isn't in our usual practice of naming errors so it appears as though it is for something else than just a normal error check... which is all this is.

Reply:
I agree in this case its good to rename this as err. 
But I dont believe we have hard rule on this tho? because F.E if we name the error as err in this line, which needs to be verified on line https://github.com/rancher/shepherd/blob/main/extensions/rbac/verify.go#L220 will be lost because we have another err in line https://github.com/rancher/shepherd/blob/main/extensions/rbac/verify.go#L216

Reply:
> IMO 85 - 90 should be put in a helper function. Its reused in these tests and appears to be 'create the initial user'.

In this line, we are adding users to the cluster Roles and for it to function properly we need to relogin.. this is better suited to be an action as this will be required for other tests also.